### ISSUE TITLE
Key Error on tag check policy for provisioned dynamodb  

### ISSUE THREAD

**Issue by amitsehgal** [cloud-custodian/cloud-custodian#6224]
**Describe the bug**
I have a policy to check mandatory tag in config-rule mode. The policy works for on-demand pricing but not for provisioned pricing and throws key exception.
[ERROR] KeyError: 'BillingModeSummary'
Traceback (most recent call last):
  File "/var/task/custodian_policy.py", line 4, in run
    return handler.dispatch_event(event, context)
  File "/var/task/c7n/handler.py", line 166, in dispatch_event
    p.push(event, context)
  File "/var/task/c7n/policy.py", line 1138, in push
    return mode.run(event, lambda_ctx)
  File "/var/task/c7n/policy.py", line 854, in run
    resources = super(ConfigRuleMode, self).run(event, lambda_context)
  File "/var/task/c7n/policy.py", line 438, in run
    resources = self.resolve_resources(event)
  File "/var/task/c7n/policy.py", line 836, in resolve_resources
    return [source.load_resource(self.cfg_event['configurationItem'])]
  File "/var/task/c7n/resources/dynamodb.py", line 25, in load_resource
    if 'LastUpdateToPayPerRequestDateTime' in resource['BillingModeSummary']:

**To Reproduce**
Create a policy to tag present with config-rule mode
**Expected behavior**
mark-for-op

**Background (please complete the following information):**
 - OS: linux
 - Python Version: [e.g. python 3.8.1]
 - Custodian Version: 0.97.0
 - Tool Version: [if applicable]
 - Cloud Provider: [e.g. gcp, aws, azure]
 - Policy: [please exclude any account/sensitive information]
```yaml
vars:
  config-mode: &config-mode
    type: config-rule
    role: arn:aws:iam::xxxxx:role/role
 - name: my-policy-dynamodb-mandatory-tag
    resource: dynamodb-table
    description: |
       The policy checks the dynamodb table required tags
    mode: *config-mode
    filters:
      - or: *filters-absent
    actions:
      - <<: *post-finding-failed
        title: "dynamodb-mandatory-tags-missing"
        description: "Mandatory tags are required on dynamodb"
      - type: mark-for-op
        op: delete
        hours: 2
      - <<: *notify
        violation_desc: "Mandatory Tags are missing on Dynamodb table
```
 - Traceback: [if applicable, please exclude sensitive/account information]
 - `custodian version --debug` output

**Additional context**

**Comment by kapilt** [comment]
you mentioned your policy was a config-rule execution-mode but thats not reflected in the policy example in the issue.

**Comment by amitsehgal** [comment]
my bad....i have updated the policy

**Comment by kapilt** [comment]
this is fixed, although it needed a new fix for the latest release that will be in 0.9.11
